vroxy.py

```diff
@@ -110,117 +19,55 @@ async def get(self):
-        else:
-            # use either the given user sort, or extrapolate for the given preset mode
-            s = self.request.query.get("s")
-            if (s): sort = s.replace(" ", "").split(",")
-            else: sort = list(sort_opts[mode])
-            try: 
-                if host.index("vimeo") > -1:
-                    sort.append("proto:m3u8_native")
-            except: pass
-            ytdl_opts["format_sort"] = sort
-            cacheId = ",".join(sort)
-
-        _id = f"{cacheId}~{url}"
-        if _id in cache_map:
-            item = cache_map[_id]
-            if item.expiry < curTime:
-                print(f"[{rid}] Cache expired")
-                del cache_map[_id]
-            else:
-                # wait until the other request for the same url resolves,
-                # then use the cached url from that
-                while item.processing:
-                    await sleep(1)
-                print(f"[{rid}] Resolving '{cacheId}' for url: {url}")
-                print(f"[{rid}] Cache hit")
-                print(item.resolved_url)
-                print(
-                    f"[{rid}] {item.resolved_format} expires in {item.expiry - curTime} seconds", flush=True
-                )
-                item.lastAccess = curTime
-                return item.resolved_url or ""
-
-        cache_map[_id] = item = Item(url, sort)
-
-        # wait for an pool slot to open
-        timeout = curTime + 30  # 30 seconds timelimit for waiting
-        while pool.count >= pool_max:
-            if curTime > timeout: return None
-            await sleep(1)
-        with YoutubeDL(ytdl_opts) as ytdl:
-            print(f"[{rid}] Resolving '{cacheId}' for url: {url}")
-            print(f"[{rid}] Fetching fresh info", flush=True)
-            pool.add()
-            result = ytdl.extract_info(url, download=False)
-            # print(result.keys())
-            item.resolve(result)
-            pool.remove()
-            print(f"[{rid}] {item.resolved_url}")
-            print(f"[{rid}] {item.resolved_format} expires in {item.expiry - curTime} seconds", flush=True)
-
-        item.lastAccess = curTime
-        return item.resolved_url or ""
-
+        url = None
+        try:
+            url = await resolveUrl(self.request.query)
+            res = web.Response(status=307, headers={"Location": url})
+        except Error400BadRequest:
+            res = web.Response(status=400)
+        except Error403Forbidden:
+            res = web.Response(status=403)
+        except Error403Whitelist:
+            res = web.Response(status=403, text="Domain not in whitelist")
+        except Error404NotFound:
+            res = web.Response(status=404)
+        except Error408Timeout:
+            res = web.Response(status=408)
+        except Error410Gone:
+            res = web.Response(status=410)
+        except Error429TooManyRequests:
+            res = web.Response(status=429)
+        except:
```

Comment:
This bare except here is too broad and will cause problems with ctrl+c https://peps.python.org/pep-0008/#:~:text=A%20bare%20except%3A%20clause%20will%20catch%20SystemExit%20and%20KeyboardInterrupt%20exceptions